lib/generate-loader/loader-generator.js

```diff
@@ -0,0 +1,103 @@
+const path = require('path');
+const mkdirp = require('mkdirp');
+const _ = require('lodash');
+const Generator = require('yeoman-generator');
+
+/**
+ * Formats a string into webpack loader format
+ * (eg: 'style-loader', 'raw-loader')
+ *
+ * @param {string} name A loader name to be formatted
+ * @returns {string} The formatted string
+ */
+function makeLoaderName(name) {
+	name = _.kebabCase(name);
+	if (!/loader$/.test(name)) {
+		name += '-loader';
+	}
+	return name;
+}
+
+/**
+ * A yeoman generator class for creating a webpack
+ * loader project. It adds some starter loader code
+ * and runs `webpack-defaults`.
+ *
+ * @class LoaderGenerator
+ * @extends {Generator}
+ */
+class LoaderGenerator extends Generator {
+	prompting() {
+		const prompts = [
+			{
+				type: 'input',
+				name: 'name',
+				message: 'Loader name',
+				default: 'my-loader',
+				filter: makeLoaderName,
+				validate: str => str.length > 0,
+			},
+		];
+
+		return this.prompt(prompts).then(props => {
+			this.props = props;
+		});
+	}
+
+	default() {
+		if (path.basename(this.destinationPath()) !== this.props.name) {
+			this.log(
+				'Your loader must be inside a folder named ' +
+				this.props.name +
+				'\nI will create this folder for you.'
+			);
+			mkdirp(this.props.name);
+			this.destinationRoot(this.destinationPath(this.props.name));
```

Comment:
Store `destinationPath(this.props.name)` in a variable for new contributors to easier know what it is